test/generate-json.py

```diff
@@ -0,0 +1,177 @@
+        payload {str} -- Payload content
+        destination_device_id {int} -- Id of the device that will receive the message
+
+    Returns:
+        str -- bytecode message
+    """
+
+    #Needed to build header
+    payload_size = struct.calcsize(message['format'])
+
+    #Create and pack header
+    header = [b'B', b'R', payload_size, message_id, 0, destination_device_id]
+    final_header = struct.pack('<ccHHBB', *header)
+
+    #Create Checksum
+    checksum = calculate_protocol_checksum(final_header, payload)
+    final_checksum = struct.pack('<H', checksum)
+    return b''.join([final_header, payload, final_checksum])
+
+
+def main():
+    """Main function
+    Run the main script
+    """
+
+    # argparse for future devices
+    parser = argparse.ArgumentParser(
+        description="Script to generate json test description file for ping protocol!")
+    parser.add_argument(
+        '--device',
+        action="store",
+        type=str,
+        default="Ping1D",
+        help="Ping device type.")
+    args = parser.parse_args()
+
+    # Create json file with everything that we need
+    json_content = {
+        'device': args.device,
+        'messages': []
+    }
+    for message_id, ping_message in Message.payloadDict.items():
+        data = {
+            'name': ping_message['name'],
+            'id': message_id,
+            'data': None,
+            'bytes': None
+        }
+        var_payload = {}
+        key = ''
+        # Generate random text or array of bytes
+        if message_id in [Message.PING1D_ASCII_TEXT, Message.PING1D_NACK]:
+            if message_id == Message.PING1D_ASCII_TEXT:
+                key = 'msg'
+            elif message_id == Message.PING1D_NACK:
+                key = 'nack_msg'
+            var_payload[key] = b' !"#$%&\'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_`abcdefghijklmnopqrstuvwxyz{|}~\0'
+        elif message_id == Message.PING1D_PROFILE:
+            var_payload['data'] = [x for x in range(200)]
+
+        # Create data with all bytes of the message
+        payload = generate_fake_data(ping_message['format'])
+        if var_payload:
+            size = 0
+            for key, value in var_payload.items():
+                payload += struct.pack(str(len(value)) + 'B', *value)
+                size += len(value)
+            ping_message['format'] = ping_message['format'] + str(size) + 'B'
+
+        data['data'] = generate_payload_dict(ping_message['format'],
+                                             payload, ping_message['field_names'])
+        if key:
+            if isinstance(var_payload[key], bytes):
+                data['data'][key] = var_payload[key].decode("ascii")
+            else:
+                data['data'][key] = var_payload[key]
+
+        data['bytes'] = list(pack_protocol_message(message_id, ping_message, payload, 1))
+        json_content['messages'].append(data)
+
+    with open('ping1D_v3.17.json', 'w') as output_file:
```

Comment:
is this filename relevant?

Reply:
This is the name of the file used by the parser to generate test.cpp.